C error on generic fn call from another generic fn

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.11 16c9bf2, press to see full `v doctor` output</summary>

|V full version      |V 0.4.11 efe611ed9937f8553b037710c86614fcd9b44f83.16c9bf2
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.6.8-arch1-1 (linux@archlinux) (gcc (GCC) 13.2.1 20230801, GNU ld (GNU Binutils) 2.41.0) #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000
|Processor           |16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P
|Memory              |2.73GB/15.35GB
|                    |
|V executable        |/home/ge/.vlang/v
|V last modified time|2025-09-17 23:07:15
|                    |
|V home dir          |OK, value: /home/ge/.vlang
|VMODULES            |OK, value: /home/ge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /tmp
|                    |
|Git version         |git version 2.50.0
|V git status        |weekly.2025.16-790-g16c9bf2f
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.1.1 20250425
|gcc version         |gcc (GCC) 15.1.1 20250425
|clang version       |clang version 20.1.6
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.41

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  generic_fn_call.v && generic_fn_call`
```v
fn fake_prompt[T](msg string, default ?T) string {
	return build_prompt(msg, default)
}

fn build_prompt[T](msg string, default ?T) string {
	mut s := msg
	if default != none {
		s += ' [${default}]'
	}
	s += ': '
	return s
}

fn main() {
	p1 := fake_prompt[string]('hello', none)
	p2 := fake_prompt[string]('hello', 'world')
	assert p1 == 'hello: '
	assert p2 == 'hello [world]: '
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/generic_fn_call.01K5CYYZWMND9JK2TDQYMGM3AZ.tmp.c:5263: error: cannot convert 'struct string' to 'struct _option_string'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Successful compilation.

Replacing `return build_prompt(msg, default)` with `return build_prompt[T](msg, default)` causes another error, but code compiles with `-no-skip-unused`.

---

In additional context this code compiles only with `-no-skip-unused`:

<details><summary>Details</summary>
<p>

```v
fn build_prompt[T](msg string, default ?T) string {
	mut s := msg
	if default != none {
		s += ' [${default}]'
	}
	s += ': '
	return s
}

fn main() {
	p1 := build_prompt[string]('hello', none)
	p2 := build_prompt[string]('hello', 'world')
	assert p1 == 'hello: '
	assert p2 == 'hello [world]: '
}
```
With `-skip-unused`:
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/generic_fn_call.01K5CZF5Z0312SKKDZPEKDW6BZ.tmp.c:5271: warning: implicit declaration of function 'builtin___option_ok'
cc: /home/ge/.vlang/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
cc: tcc: error: undefined symbol 'builtin___option_ok'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .

```

</p>
</details>

--- a/vlib/v/gen/c/str_intp.v
+++ b/vlib/v/gen/c/str_intp.v
@@ -225,6 +225,9 @@ fn (mut g Gen) str_val(node ast.StringInterLiteral, i int, fmts []u8) {
 					if cast_sym.info is ast.Aggregate {
 						exp_typ = cast_sym.info.types[g.aggregate_type_idx]
 					}
+					if exp_typ.has_flag(.option) && expr.obj.is_unwrapped {
+						exp_typ = exp_typ.clear_flag(.option)
+					}
 				}
 			}
 		}

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -716,6 +716,9 @@ fn (mut w Walker) expr(node_ ast.Expr) {
 					}
 				}
 			}
+			if node.obj is ast.Var && node.obj.is_unwrapped {
+				w.used_option++
+			}
 			w.or_block(node.or_expr)
 		}
 		ast.LambdaExpr {

--- a/vlib/v/tests/options/option_unwrap_none_test.v
+++ b/vlib/v/tests/options/option_unwrap_none_test.v
@@ -1,0 +1,19 @@
+fn fake_prompt[T](msg string, default ?T) string {
+	return build_prompt(msg, default)
+}
+
+fn build_prompt[T](msg string, default ?T) string {
+	mut s := msg
+	if default != none {
+		s += ' [${default}]'
+	}
+	s += ': '
+	return s
+}
+
+fn test_main() {
+	p1 := fake_prompt[string]('hello', none)
+	p2 := fake_prompt[string]('hello', 'world')
+	assert p1 == 'hello: '
+	assert p2 == 'hello [world]: '
+}

